src/util/triplea/MapXMLCreator/TerritoryConnectionsPanel.java

```diff
@@ -0,0 +1,256 @@
+import java.awt.Graphics;
+import java.awt.Point;
+import java.awt.Polygon;
+import java.awt.Shape;
+import java.awt.event.ActionEvent;
+import java.awt.event.MouseEvent;
+import java.awt.geom.Area;
+import java.awt.geom.Rectangle2D;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import javax.swing.AbstractAction;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.SwingUtilities;
+
+import util.image.ConnectionFinder;
+
+/**
+ * 
+ * @author Erik von der Osten
+ * 
+ */
+public class TerritoryConnectionsPanel extends ImageScrollPanePanel
+{
+	
+	String m_selectedTerritory = null;
+
+	private TerritoryConnectionsPanel()
+	{
+		s_instance = this;
+	}
+	
+	public static void layout(final MapXMLCreator mapXMLCreator, final JPanel stepActionPanel)
+	{
+		s_mapXMLCreator = mapXMLCreator;
+		final TerritoryConnectionsPanel panel = new TerritoryConnectionsPanel();
+		panel.layout(stepActionPanel);
+		mapXMLCreator.setAutoFillAction(new AbstractAction()
+		{
+			private static final long serialVersionUID = -8508734371454749752L;
+
+			@Override
+			public void actionPerformed(ActionEvent arg0)
+			{
+				panel.paintPreparation(null);
+				panel.repaint();
+			}
+		});
+	}
+
+	protected void paintCenterSpecifics(final Graphics g, final String centerName, final FontMetrics fontMetrics, final Point item, final int x_text_start)
+	{
+		if (centerName.equals(m_selectedTerritory))
+		{
+			final Rectangle2D stringBounds = fontMetrics.getStringBounds(centerName, g);
+			g.setColor(Color.yellow);
+			g.fillRect(Math.max(0, x_text_start - 2), Math.max(0, item.y - 6), (int) stringBounds.getWidth() + 4, (int) stringBounds.getHeight());
+			g.setColor(Color.red);
+			g.drawString(centerName, Math.max(0, x_text_start), item.y + 5);
+		}
+		g.setColor(Color.red);
+	}
+	
+	protected void paintPreparation(final Map<String, Point> centers)
+	{
+		if (centers != null && !MapXMLHelper.s_territoyConnections.isEmpty())
+			return;
+		final Map<String, List<Area>> territoryAreas = new HashMap<String, List<Area>>();
+		for (final String territoryName : s_polygons.keySet())
+		{
+			final List<Polygon> listOfPolygons = s_polygons.get(territoryName);
+			final List<Area> listOfAreas = new ArrayList<Area>();
+			for (final Polygon p : listOfPolygons)
+			{
+				listOfAreas.add(new Area(p));
+			}
+			territoryAreas.put(territoryName, listOfAreas);
+
+		}
+		final String lineWidth = JOptionPane.showInputDialog(null, "Enter the width of territory border lines on your map? \r\n(eg: 1, or 2, etc.)", "1");
+		if (lineWidth == null)
+			return;
+		int scalePixels = 8;
+		double minOverlap = 32;
+		try
+		{
+			final int lineThickness = Integer.parseInt(lineWidth);
+			scalePixels = lineThickness * 4;
+			minOverlap = scalePixels * 4;
+		} catch (final NumberFormatException ex)

```

Comment:
Better yet, move the Integer.parseInt to a helper method, since this logic is done repeatedly.